File: coding/Leetcode/task_1920.cpp
// https://leetcode.com/problems/build-array-from-permutation/
// Created by ruslan515 on 14.01.23.
//

#include <iostream>
#include <vector>
#include <stdlib.h>

using namespace std;

class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> ans;
        for (int num: nums) {
            ans.push_back(nums[num]);
        }
        return ans;
    }
};

void read_data(vector<int>& nums) {
    int t;
    while (cin >> t)
        nums.push_back(t);
    return;
}

int main() {
    vector<int> nums;
    read_data(nums);
    Solution solve;
    for(int i: solve.buildArray(nums)) {
        cout << i << " ";
    }
    return 0;
}

File: coding/Leetcode/Tasks/task_1920.java
package leetcode.tasks;

public class task_1920 {
    class Solution {
        public int[] buildArray(int[] nums) {
            int n = nums.length;
            int[] answer = new int[n];
            for (int i = 0; i < n; i++) {
                int idx = nums[i];
                answer[i] = nums[idx];
            }

            return answer;
        }
    }

    public static void main(String[] args) {
        int[] nums;
        int[] x;
        int[] answer;

        Solution solution = new task_1920().new Solution();

        nums = new int[]{0, 2, 1, 5, 3, 4};
        answer = new int[]{0, 1, 2, 4, 5, 3};
        x = solution.buildArray(nums);
        assert answer == x : "Answer is different";

        nums = new int[]{0, 2, 1, 5, 3, 4};
        answer = new int[]{0, 1, 2, 4, 5, 3};
        x = solution.buildArray(nums);
        assert answer == x : "Answer is different";

    }
}
